homeassistant/components/airthings_ble/sensor.py

```diff
@@ -0,0 +1,185 @@
+        native_unit_of_measurement=VOLUME_BECQUEREL,
+        name="Radon 1-day average",
+        state_class=SensorStateClass.MEASUREMENT,
+        icon="mdi:radioactive",
+    ),
+    "radon_longterm_avg": SensorEntityDescription(
+        key="radon_longterm_avg",
+        native_unit_of_measurement=VOLUME_BECQUEREL,
+        name="Radon longterm average",
+        state_class=SensorStateClass.MEASUREMENT,
+        icon="mdi:radioactive",
+    ),
+    "radon_1day_level": SensorEntityDescription(
+        key="radon_1day_level",
+        name="Radon 1-day level",
+        icon="mdi:radioactive",
+    ),
+    "radon_longterm_level": SensorEntityDescription(
+        key="radon_longterm_level",
+        name="Radon longterm level",
+        icon="mdi:radioactive",
+    ),
+    "temperature": SensorEntityDescription(
+        key="temperature",
+        device_class=SensorDeviceClass.TEMPERATURE,
+        native_unit_of_measurement=TEMP_CELSIUS,
+        name="Temperature",
+    ),
+    "humidity": SensorEntityDescription(
+        key="humidity",
+        device_class=SensorDeviceClass.HUMIDITY,
+        native_unit_of_measurement=PERCENTAGE,
+        name="Humidity",
+    ),
+    "pressure": SensorEntityDescription(
+        key="pressure",
+        device_class=SensorDeviceClass.PRESSURE,
+        native_unit_of_measurement=PRESSURE_MBAR,
+        name="Pressure",
+    ),
+    "battery": SensorEntityDescription(
+        key="battery",
+        device_class=SensorDeviceClass.BATTERY,
+        native_unit_of_measurement=PERCENTAGE,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        name="Battery",
+    ),
+    "co2": SensorEntityDescription(
+        key="co2",
+        device_class=SensorDeviceClass.CO2,
+        native_unit_of_measurement=CONCENTRATION_PARTS_PER_MILLION,
+        name="co2",
+    ),
+    "voc": SensorEntityDescription(
+        key="voc",
+        device_class=SensorDeviceClass.VOLATILE_ORGANIC_COMPOUNDS,
+        native_unit_of_measurement=CONCENTRATION_PARTS_PER_BILLION,
+        name="VOC",
+        icon="mdi:cloud",
+    ),
+    "illuminance": SensorEntityDescription(
+        key="illuminance",
+        device_class=SensorDeviceClass.ILLUMINANCE,
+        native_unit_of_measurement=LIGHT_LUX,
+        name="Illuminance",
+    ),
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: config_entries.ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Airthings BLE sensors."""
+    is_metric = hass.config.units.is_metric
+
+    coordinator: DataUpdateCoordinator[AirthingsDevice] = hass.data[DOMAIN][
+        entry.entry_id
+    ]
+
+    # we need to change some units
+    sensors_mapping = SENSORS_MAPPING_TEMPLATE.copy()
+    if not is_metric:
+        for val in sensors_mapping.values():
+            if val.native_unit_of_measurement is not VOLUME_BECQUEREL:
+                continue
+            val.native_unit_of_measurement = VOLUME_PICOCURIE
```

Comment:
Please don't modify the entity descriptions. Pass the unit separately to the entity or include both variants of the affected descriptions with an extra id for `is_metric` in the entity description container key.